extension/ezoe/modules/ezoe/upload.php

```diff
@@ -167,10 +174,23 @@
                         $attr->store();
                         break;
                     case 'ezimage':
-                        // validation has been done by eZContentUpload
                         $content = $attr->attribute( 'content' );
-                        $content->setAttribute( 'alternative_text', $postVar );
-                        $content->store( $attr );
+
+                        // Check if the alt text is required
+                        if(
+                            !trim( $postVar ) &&
+                            eZImageType::isAltTextRequired( $attr )
+                        )
+                        {
+                            throw new InvalidArgumentException(
+                                ezpI18n::tr( 'design/standard/error/kernel','Alternative text required.' )
+                            );
+                        }
+                        else
+                        {
+                            $content->setAttribute( 'alternative_text', trim( $postVar ) );
```

Comment:
Should I be merging isAltTextRequired into validateImageAltText.  This would mean that I'm going to have to return a string from validateImageAltText that indicates whether or not the field is required, etc.  I kept them separate for clarity and also for future preparation for more elaborate validation.